### ISSUE TITLE
tcolorbox and TikZ external

### ISSUE THREAD

**Issue by thefiercerabbit** [T-F-S/tcolorbox#71]
## Description
I have the following in my preambule:
```
\usepackage[most]{tcolorbox}
\usepackage{tikz}
\usetikzlibrary{external}
\tikzexternalize
\tcbset{shield externalize} 
```

When creating a `tcolorbox` with the skin family `enhanced` and using `attach boxed title` option, then I got an error from `TikZ` trying to externalize the title box (even though `shield externalize` is set).

## Example
```
\documentclass[a4paper]{article} 
\usepackage[most]{tcolorbox}
\usepackage{tikz}
\usetikzlibrary{external}
\tikzexternalize
\tcbset{shield externalize} 

\begin{document}

\begin{tcolorbox}[enhanced, attach boxed title to top left, title=Title]
    Test
\end{tcolorbox}

\end{document}
```

#### Output:
```
! Extra \else.
<argument> ...r}}]{\box \tcb@titlebox }}\fi \else 
                                                  \tcb@hasTitletrue \tcbdimt...
l.10 ...tach boxed title to top left, title=Title]
```

### Other
The issue is also referenced here: https://tex.stackexchange.com/questions/486102/problems-with-tcolorbox-and-tikz-external

**Comment by T-F-S** [comment]
The reason for the problem is that the boxed title is a `tcolorbox` itself, but with a higher layer number. For this higher layer, the global setting of `shield externalize` is reset to the default, i.e. to be `false`.

For almost all options this reset is useful (documented in section 4.16). But, I think, for `shield externalize` this is not true. Also, the documentation claims that `\tcbset{shield externalize}` in the preamble shields any `tcolorbox` inside the document.

The problem should vanish, if you replace `\tcbset{shield externalize}` by `\tcbsetforeverylayer{shield externalize}`, because this is applied to all boxed on all layers:

```latex
\documentclass[a4paper]{article}
\usepackage[most]{tcolorbox}
\usepackage{tikz}
\usetikzlibrary{external}
\tikzexternalize
%\tcbset{shield externalize}
\tcbsetforeverylayer{shield externalize}% <--- interim solution before bug fix


\begin{document}

\begin{tcolorbox}[enhanced, attach boxed title to top left, title=Title]
    Test
\end{tcolorbox}

\end{document}

```
Since I cannot imagine a use case, where one does not want to apply this for all layers, I will change the implementation for the next version of `tcolorbox`. The future change will remove `shield externalize` from the `reset` operation. After that, `\tcbset{shield externalize}` will be adequate as claimed by the documentation.

**Comment by T-F-S** [comment]
Resolved for version 4.21 (2019-09-19)
